### PR TITLE
707 add visr b01 1 as a beamline with devices

### LINKED ISSUE #707: Add ViSR (b01-1) as a beamline with devices
Devices needed
- aravisdetector instance for manta (webcam 2)
- some detector for mako (webcam 1)
- motor device for the moving mirror (s)
- stage device for the 3d stage https://github.com/DiamondLightSource/ViSR/issues/3

### PR DESCRIPTION
**Pull request by stan-dot**
Fixes #707

### Instructions to reviewer on how to test:
1. make sure `dodal connect b-01-1` is successful
2. panda and manta should connect (skipping the stage for now)

### Checks for reviewer
- [ ] Would the PR title make sense to a scientist on a set of release notes
- [ ] If a new device has been added does it follow the [standards](https://github.com/DiamondLightSource/dodal/wiki/Device-Standards)
- [ ] If changing the API for a pre-existing device, ensure that any beamlines using this device have updated their Bluesky plans accordingly
- [ ] Have the connection tests for the relevant beamline(s) been run via `dodal connect ${BEAMLINE}`

### CODE REVIEW

**Comment by stan-dot** [comment]
this is still not connecting :(

File: src/dodal/beamlines/b01-1.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+#     If this is called when already instantiated in i03, it will return the existing object.
+#     """
+#     return device_instantiation(
+#         Synchrotron,
+#         "synchrotron",
+#         "",
+#         wait_for_connection,
+#         fake_with_ophyd_sim,
+#         bl_prefix=False,
+#     )
```
Comment (Relm-Arrowny) [inline src/dodal/beamlines/b01-1.py:52]: I think you can remove this, I am guessing visr will never need a synchrotron device. As there is no real control of the "front end".

Comment (Relm-Arrowny) [reply src/dodal/beamlines/b01-1.py:52]: I cannot connect to it on my work station, I am guessing I have to be in the visr network to connect?

File: src/dodal/beamlines/b01-1.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+from ophyd_async.panda import HDFPanda
+
+from dodal.common.beamlines.beamline_utils import (
+    device_instantiation,
+)
+from dodal.common.beamlines.beamline_utils import set_beamline as set_utils_beamline
+from dodal.log import set_beamline as set_log_beamline
+
+# BL = get_beamline_name("BL01C")
+BL = "c01"
```
Comment (DiamondJoseph) [inline src/dodal/beamlines/b01-1.py:12]:
```suggestion
BL = get_beamline_name("c01")
```

File: src/dodal/beamlines/b01-1.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+set_log_beamline(BL)
+set_utils_beamline(BL)
+
+
+static_directory_provider = StaticDirectoryProvider("/tmp/bluesky_test_static")
+# set_directory_provider(PandASubdirectoryProvider())
+
```
Comment (DiamondJoseph) [inline src/dodal/beamlines/b01-1.py:19]:
```suggestion
set_directory_provider(StaticDirectoryProvider("/dls/b01-1/data/"))
```

File system seems to be hiccuping (again) but when it's recovered find a sensible visit dir for this beamline

File: src/dodal/beamlines/b01-1.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+def panda(
+    wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
+) -> HDFPanda:
+    """Get the i03 panda_fast_grid_scan device, instantiate it if it hasn't already been.
+    If this is called when already instantiated in i03, it will return the existing object.
+    This is used instead of the zebra_fast_grid_scan device when using the PandA.
+    """
```
Comment (DiamondJoseph) [inline src/dodal/beamlines/b01-1.py:27]: not relevant to this beamline

File: src/dodal/beamlines/b01-1.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+BL = "c01"
+set_log_beamline(BL)
+set_utils_beamline(BL)
+
+
+static_directory_provider = StaticDirectoryProvider("/tmp/bluesky_test_static")
+# set_directory_provider(PandASubdirectoryProvider())
+
+
+def panda(
```
Comment (DiamondJoseph) [inline src/dodal/beamlines/b01-1.py:21]: Is there a panda on this beamline? I don't see it on the beamline synoptic screen, nor is there a soft-ioc for it in kubernetes

File: src/dodal/beamlines/b01-1.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        device_factory=HDFPanda,
+        name="panda",
+        prefix="-EA-PANDA-01:",
+        wait=wait_for_connection,
+        fake=fake_with_ophyd_sim,
+        directory_provider=static_directory_provider,
+    )
+
+
+# @skip_device(lambda: BL == "s03")
```
Comment (DiamondJoseph) [inline src/dodal/beamlines/b01-1.py:38]:
```suggestion
```

i03 specific

File: src/dodal/beamlines/b01-1.py (hunk 1 above)
Comment (DiamondJoseph) [reply src/dodal/beamlines/b01-1.py:52]: I think unskip it and uncomment it, should be able to see it from the beamline network and it gives us a baseline signal for examples

File: src/dodal/beamlines/b01-1.py [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+#     )
+
+
+def manta(
+    wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
+) -> AravisDetector:
+    return device_instantiation(
+        AravisDetector,
+        "manta",
+        "-DI-DCAM-02:",
```
Comment (DiamondJoseph) [inline src/dodal/beamlines/b01-1.py:61]: this device is disconnected on the synoptic for me, but `-DI-DCAM-03:` is a Manta that is present?

**Review by DiamondJoseph — DISMISSED** [review]

**Comment by DiamondJoseph** [comment]
Realised I was looking at BL01B: is VISR 01B or 01C? There's no synoptic screen for the latter.

File: src/dodal/beamlines/b01-1.py (hunk 7 above)
Comment (stan-dot) [reply src/dodal/beamlines/b01-1.py:61]: it should be and Marcell is doing the IOCs into argocd migration this week afaik

**Comment by stan-dot** [comment]
@DiamondJoseph , ViSR is the C variant and it does not have an epics screen

File: src/dodal/beamlines/b01-1.py (hunk 1 above)
Comment (stan-dot) [reply src/dodal/beamlines/b01-1.py:52]: uncommented and unskipped as a baseline signal

File: src/dodal/beamlines/b01-1.py (hunk 2 above)
Comment (stan-dot) [reply src/dodal/beamlines/b01-1.py:12]: done

File: src/dodal/beamlines/b01-1.py (hunk 3 above)
Comment (stan-dot) [reply src/dodal/beamlines/b01-1.py:19]: is the file system gaslighting me at the moment?

image

File: src/dodal/beamlines/b01-1.py (hunk 4 above)
Comment (stan-dot) [reply src/dodal/beamlines/b01-1.py:27]: deleted

File: src/dodal/beamlines/b01-1.py (hunk 5 above)
Comment (stan-dot) [reply src/dodal/beamlines/b01-1.py:21]: yes, there is. there is aan argocd migration going on at the moment.

@marcelldls mentioned it shouldn't take too long

File: src/dodal/beamlines/b01-1.py (hunk 6 above)
Comment (stan-dot) [reply src/dodal/beamlines/b01-1.py:38]: of course, deleted

File: src/dodal/beamlines/b01-1.py (hunk 3 above)
Comment (DiamondJoseph) [reply src/dodal/beamlines/b01-1.py]: I think someone needs to raise a ticket because it's been consistently like that for days.

**Comment by stan-dot** [comment]
panda is supposed to be off at the moment

**Comment by stan-dot** [comment]
trying to fix manta connect errors, pending comment from @marcelldls

**Comment by stan-dot** [comment]
connects ok

File: src/dodal/beamlines/b01-1.py (hunk 3 above)
Comment (stan-dot) [reply src/dodal/beamlines/b01-1.py]: made a ticket https://jira.diamond.ac.uk/servicedesk/customer/portal/4/ITHD-4851

File: src/dodal/beamlines/b01-1.py (hunk 7 above)
Comment (stan-dot) [reply src/dodal/beamlines/b01-1.py]: argocd ready https://argocd.diamond.ac.uk/applications?showFavorites=false&proj=&sync=&autoSync=&health=&namespace=&cluster=&labels=

**Comment by stan-dot** [comment]
@DominicOram is the use of the force override for the non-standard beamline name ok here?

**Comment by DominicOram** [comment]
> @DominicOram is the use of the force override for the non-standard beamline name ok here?

Really `get_beamline_name` is just an abstraction around getting the `BEAMLINE` env variable. So why do you need to force it? Is there no `BEAMLINE` env defined for VISR? Or you want to be able to connect to VISR from other beamlines?

**Comment by stan-dot** [comment]
@DominicOram this is quite ready now

File: src/dodal/beamlines/b01_1.py [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+)
+
+"""
+NOTE: Due to ArgoCD and the k8s cluster configuration those PVs are not available remotely.
+You need to be on the beamline-local network to access them.
+The simplest way to do this is to `ssh b01-1-ws001` and run `dodal connect b01_1` from there.
+remember about the underscore in the beamline name.
+
+See the IOC status here:
+https://argocd.diamond.ac.uk/applications?showFavorites=false&proj=&sync=&autoSync=&health=&namespace=&cluster=&labels=
```
Comment (DominicOram) [inline src/dodal/beamlines/b01_1.py:35]: Should: This link doesn't work for me, I just see an empty screen in argo. Do I need specific permissions?

**Review by DominicOram — APPROVED** [review]
Great, one question otherwise good.

File: src/dodal/beamlines/b01_1.py (hunk 8 above)
Comment (stan-dot) [reply src/dodal/beamlines/b01_1.py:35]: I don't think so, it just worked for me when I log in with CAS.

Maybe I was added to the allowlist though

Comment (stan-dot) [reply src/dodal/beamlines/b01_1.py:35]: yes, it's the same as permissions for the cluster

Comment (DominicOram) [reply src/dodal/beamlines/b01_1.py:35]: Ok, maybe add a note to that affect then

**Pull request merged**